lib/auth/notifier/auth_state_notifier.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../authenticator.dart';
+import '../model/auth_result.dart';
+import '../model/auth_state.dart';
+import '../user_info_storage.dart';
+
+class AuthStateNotifier extends StateNotifier<AuthState> {
+  AuthStateNotifier() : super(const AuthState.unknown()) {
+    if (_authenticator.isAlreadySignedIn) {
+      state = AuthState(
+        result: AuthResult.success,
+        isLoading: false,
+        userId: _authenticator.userId,
+      );
+    }
+  }
+
+  final _authenticator = const Authenticator();
+  final _userInfoStorage = const UserInfoStorage();
+
+  Future<void> logOut() async {
+    state = state.copiedWithIsLoading(true);
+    await _authenticator.signOut();
+    state = const AuthState.unknown();
+  }
+
+  Future<void> loginWithGoogle() async {
+    state = state.copiedWithIsLoading(true);
+    final result = await _authenticator.signInWithGoogle();
+    final userId = _authenticator.userId;
+    if (result == AuthResult.success && userId != null) {
+      final savingUserInfoToFirestoreWasSuccessful =
+          await _userInfoStorage.saveUserInfo(
+        userId: userId,
+        displayName: _authenticator.displayName,
+        email: _authenticator.email,
+      );
+      if (savingUserInfoToFirestoreWasSuccessful) {
+        state = AuthState(
+          result: result,
+          isLoading: false,
+          userId: userId,
+        );
+      } else {
+        state = const AuthState.unknown();
+      }
+    } else {
+      state = state.copiedWithIsLoading(false);
+    }
```

Comment:
moze byc to kwestia gustu ale jak w ifie zmieniasz tylko jedną zmienną to może lepiej na ternary przerzucić

Reply:
Okej!